packages/flutter/lib/src/widgets/overscroll_indicator.dart

```diff
@@ -637,18 +644,279 @@ class _GlowingOverscrollIndicatorPainter extends CustomPainter {
+      }
+
+      assert(notification.metrics.axis == widget.axis);
+      if (_accepted) {
+        if (notification.velocity != 0.0) {
+          assert(notification.dragDetails == null);
+          _stretchController.absorbImpact(notification.velocity.abs());
+        } else {
+          assert(notification.overscroll != 0.0);
+          if (notification.dragDetails != null) {
+            _stretchController.pull(notification.overscroll.abs() / notification.metrics.viewportDimension);
+          }
+        }
+      }
+    } else if (notification is ScrollEndNotification && notification.dragDetails != null
+        || notification is ScrollUpdateNotification && notification.dragDetails != null) {
+      _stretchController.scrollEnd();
+    }
+    _lastNotification = notification;
+    return false;
+  }
+
+  @override
+  void dispose() {
+    _stretchController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return NotificationListener<ScrollNotification>(
+      onNotification: _handleScrollNotification,
+      child: AnimatedBuilder(
+        animation: _stretchController,
+        builder: (BuildContext context, Widget? child) {
+          final double stretch = _stretchController.value;
+          double x = 1.0;
+          double y = 1.0;
+          final AlignmentDirectional alignment;
+
+          switch (widget.axis) {
+            case Axis.horizontal:
+              x += stretch;
+              alignment = (_lastOverscrollNotification?.overscroll ?? 0) > 0
+                  ? AlignmentDirectional.centerEnd
+                  : AlignmentDirectional.centerStart;
+              break;
+            case Axis.vertical:
+              y += stretch;
+              alignment = (_lastOverscrollNotification?.overscroll ?? 0) > 0
+                  ? AlignmentDirectional.bottomCenter
+                  : AlignmentDirectional.topCenter;
+              break;
+          }
+
+          return Transform(
+            alignment: alignment,
+            transform: Matrix4.diagonal3Values(x, y, 1.0),
+            child: widget.child,
+          );
+        },
+      ),
+    );
+  }
+}
+
+enum _StretchState {
+  idle,
+  absorb,
+  pull,
+  recede,
+}
+
+class _StretchController extends ChangeNotifier {
+  _StretchController({ required TickerProvider vsync }) {
+    _stretchController = AnimationController(vsync: vsync)
+      ..addStatusListener(_changePhase);
+    final Animation<double> decelerator = CurvedAnimation(
+      parent: _stretchController,
+      curve: Curves.decelerate,
+    )..addListener(notifyListeners);
+    _stretchSize = decelerator.drive(_stretchSizeTween);
+  }
+
+  late final AnimationController _stretchController;
+  late final Animation<double> _stretchSize;
+  final Tween<double> _stretchSizeTween = Tween<double>(begin: 0.0, end: 0.0);
+  _StretchState _state = _StretchState.idle;
+  double _pullDistance = 0.0;
+  final double _scalar = math.e / 0.33;
```

Comment:
This can be const, is there a meaning for e / 0.33 to add as comment?

Reply:
Yes, thank you! I just updated with all of the constants from Android together. Hopefully more readable. :)